pkgs/os-specific/darwin/fix-dylib-references-in-binaries.sh

```diff
@@ -0,0 +1,47 @@
+# A setup-hook for Darwin that acts like auto patchelf: for any executable in
+# $out/bin, for any dynamic libraries it depends on which don’t have a full
+# path, this hook looks for those libraries in all this package’s dependencies
+# (and in this package’s own lib folder).
+
+# First, for every dependency that has a /lib/ dir, put it in a search path.
+fixDarwinDylibReferencesHook() {
+    addToSearchPath FIX_DYNLIB_PATH "$1/lib"
```

Comment:
I kind of free-styled this envvar name, but obviously this pollutes the entire build script with a pretty obscure name. Should we call it `NIX_...`? Any better names?

And going meta: what I'm really doing here is just accumulating a list of directories that might contain dynlibs. Surely that already exists somehow? Maybe an existing hook we can reuse? I can't imagine this is the first time anyone needs this.